README.md
# modbus_web


HISTORY
   17  vim monitor_up2.php 
   18  vim monitor_up.php 
   19  date
   20  date
   21  cd /var/www/demo/
   22  ls
   23  vim crontab.sh 
   24  vim crontab.php 
   25  vim crontab_local.php 
   26  vim crontab.sh 
   27  vim crontab_local.php 
   28  la
   29  cd
   30  cd modbus/libmodbus-3.0.3
   31  ls
   32  cd src/
   33  ls
   34  ./client_serial 1521 10
   35  cat history.txt 
   36  setinterface /dev/ttyM0 1
   37  ./client_serial 1521 10
   38  cd /var/www/demo/
   39  ls
   40  ll -a
   41  ll .*
   42  clear
   43  ll .*
   44  ll -a
   45  vim update_device.php 
   46  rm .update_device.php.swp 
   47  ll -a
   48  vim sql_dump.php 
   49  rm .sql_dump.php.swp 
   50  ll -a
   51  vim res.php 
   52  rm .res.php.swp 
   53  ll -a
   54  vim res.php 
   55  tail /var/log/mysql.log
   56  tail /var/log/mysql/mysql.log
   57  tail /var/log/mysql.err 
   58  tail /var/log/mysql.log
   59  tail /var/log/mysql.err 
   60  /etc/init.d/mysql restart
   61  /etc/init.d/mysql restart
   62  tail /var/log/mysql.err 
   63  cat test.txt 
   64  cat temp_file.txt 
   65  cat temp.txt 
   66  ls
   67  cat temp.txt 
   68  cat temp_file.txt 
   69  history 
   70  vim res.php 
   71  vim update_device.php \
   72  vim update_device.php 
   73  tail /var/log/mysql.err 
   74  cd /var/run/mysqld/
   75  ll
   76  cd ..
   77  cd /opt/
   78  ls
   79  ll
   80  cd ..
   81  cd /var/spool/
   82  ls
   83  cd cron/
   84  ls
   85  cat crontabs/
   86  cd crontabs/
   87  ls
   88  cat www-data
   89  ls
   90  cat root 
   91  ls
   92  cat www-data
   93  vim www-data
   94  cd /var/www/demo/
   95  vim crontab.php 
   96  df
   97  vim pro.html 
   98  vim sql_query.php 
   99  vim sql_query2.php 
  100  ls
  101  vim crontab.php 
  102  vim sql_dump.php 
  103  vim update_device.php 
  104  vim pro.html 
  105  vim pro.php
  106  cat temp_file.txt 
  107  vim pro.php
  108  vim pro.php
  109  vim res2.php 
  110  vim upload.php 
  111  vim download_local.php 
  112  vim ftp.php 
  113  top
  114  df
  115  ls
  116  tail /var/log/apache2/access.log
  117  tail /var/log/apache2/error.log
  118  ll
  119  rm sch 
  120  ls
  121  rm sch 
  122  cat temp_file.txt 
  123  vim temp_file.txt 
  124  serial_read3 temp_file.txt 
  125  date
  126  top
  127  cd sql
  128  mysql -uroot -p Fri\ Jun\ 26\ 03\:31\:05\ UTC\ 2015.sql 
  129  mysql -uroot -p < Fri\ Jun\ 26\ 03\:31\:05\ UTC\ 2015.sql 
  130  date 071414042015
  131  date
  132  date
  133  date
  134  date
  135  date
  136  cd ..
  137  ls
  138  rm sch 
  139  ll
  140  vim res.php 
  141  vim pro.php 
  142  cd 
  143  cd modbus/libmodbus-3.0.3/src/
  144  ls
  145  vim serial_read3.c 
  146  cd /var/www/demo/
  147  vim cancel2.php 
  148  ls
  149  rm sch
  150  php3 cancel2.php 
  151  php5 cancel2.php 
  152  vim cancel2.php 
  153  cd
  154  cd modbus/libmodbus-3.0.3/src/
  155  ls
  156  vim serial_read3.c 
  157  cd /var/www/demo/
  158  ls
  159  top
  160  php5 cancel2.php 
  161  cat cancel2.php 
  162  php5 cancel2.php 
  163  top
  164  cat temp_file.txt 
  165  cat test.txt 
  166  ls
  167  ls
  168  cp update_device.php event.php
  169  vim event.php 
  170  cd /var/www/demo/
  171  vim res.php 
  172  vim crontab_local.php 
  173  cat crontab.sh 
  174  cd sql
  175  ll
  176  ll |more 
  177  ll
  178  rm 2015-0*
  179  ll
  180  cd ..
  181  cd sql
  182  ll
  183  cd ..
  184  vim sql_dump2.php 
  185  vim sql_dump3.php 
  186  ps
  187  top
  188  vim crontab_local.php 
  189  vim crontab.php 
  190  vim sql_dump2.php 
  191  vim sql_dump3.php 
  192  vim update_device2.php 
  193  vim monitor.php 
  194  vim monitor_up.php 
  195  vim monitor_up2.php 
  196  vim res.php 
  197  ls
  198  top 
  199  top 
  200  cd /var/www/demo/
  201  vim res.php 
  202  vim download_local.php 
  203  vim res.php 
  204  vim sql_dump.php 
  205  tail /var/log/apache2/access.log
  206  cat download3.php 
  207  cat temp.txt 
  208  cat temp.txt 
  209  cat temp.txt 
  210  cat temp.txt 
  211  tail /var/log/mysql.err 
  212  /etc/init.d/mysql restart
  213  cd /var/lib/mysql/
  214  ls
  215  ll
  216  cd SystemControl/
  217  ls
  218  top
  219  ps -aux |grep mysql
  220  ps -aux
  221  ps -aux | grep mysql
  222  ps -aux | grep mysql*
  223  ps aux | grep mysql*
  224  tail /var/log/mysql.err 
  225  top
  226  /etc/init.d/mysql restart
  227  /etc/init.d/mysql restart
  228  tail /var/log/mysql.err
  229  /etc/init.d/mysql start
  230  df -h
  231  cd /var/log/
  232  ls
  233  ll
  234  cd mysql
  235  ls
  236  ll
  237  df
  238  apt-get clean
  239  apt-get autoclean
  240  df
  241  apt-get autoremove
  242  df
  243  cd /tmp/
  244  ls
  245  ll
  246  cd /var/log
  247  ll
  248  cd mysql
  249  ls
  250  ll
  251  cd mysql.log
  252  du --max-depth=1 -x | sort -n
  253  du --max-depth=1 -x
  254  du --max-depth=1 
  255  du -
  256  df --max-depth=1 -x | sort -n
  257  du
  258  cd
  259  df --max-depth=1 -x | sort -n
  260  du --max-depth=1 -x | sort -n
  261  cd ghliaw/
  262  ls
  263  ll
  264  cd libmodbus-3.0.6
  265  ll
  266  cd ..
  267  ls
  268  cd ..
  269  ls
  270  cd ghliaw/
  271  ls
  272  ll
  273  rm libmodbus-3.0.6.tar.gz 
  274  ll
  275  df
  276  cd ..
  277  ls
  278  du --max-depth=1 -x | sort -n
  279  ll
  280  ll -a
  281  pwd
  282  df
  283  du / --max-depth=1 -x | sort -n
  284  cd /usr/
  285  ls
  286  du / --max-depth=1 -x | sort -n
  287  du  --max-depth=1 -x | sort -n
  288  cd games/
  289  ls
  290  ll
  291  cd ..
  292  cd ..
  293  du /var --max-depth=1 -x | sort -n
  294  du /usr --max-depth=1 -x | sort -n
  295  cd /var/log/
  296  du /var/log --max-depth=1 -x | sort -n
  297  cd apache2/
  298  ls
  299  ll
  300  cd ..
  301  ls
  302  ll
  303  cd apache2/
  304  ls
  305  ll
  306  df
  307  df /var
  308  /etc/init.d/mysql restart
  309  /etc/init.d/mysql restart
  310  mysql -uroot -p
  311  cd /var/www/demo/
  312  ls
  313  date
  314  vim update_device.php 
  315  ll

crontab.php
<!DOCTYPE html>
<html lang="zh-TW">
<head>
	<title>優越科技測試專案</title>
		<meta charset="utf-8">
		<meta name="keywords" content="網頁關鍵字">
		 <meta name="description" content="網頁大網">
		 <link rel="stylesheet" href="css-js/layout.css">
		 <script src="css-js/jquery-1.11.0.min.js"></script>
		 <script src="css-js/layout.js"></script>
</head>
<body>
	<div id="headArea">
<div id="head-L">
<a title="優越科技" href="./index.html"><img src="images/layout_set_logo.jpg"></a>
	</div>
</div>
	 <!--head_Button_Area-->
	 <div id="headBtnArea">
		 <div id="headBtn">
				 <li id="Line">
					 <a title="資料收集" class="btn1" href="index.html">資料收集</a>
							 <div class="headBtn-items">
								 <div><a title= "新增資料" href="data_add.html">新增資料</a></div>
								 <div><a title= "程式化" href="pro.html">排程設定</a></div>
								 <!--<div id="print_sch"><a>排程列示</a></div>-->
								 <div><a title= "手動" href="res.php">排程列示</a></div>
							 </div>
				 </li>
				 <li>
					 <a title="資料庫管理" class="btn1" href="index.html">資料庫管理</a>
							 <div class="headBtn-items">
								 <div><a title= "資料庫備份" href="sql_dump.php">資料庫備份</a></div>
								 <div><a title= "排程備份" href="crontab.php">排程備份</a></div>
								 <div><a title= "資料庫操作" href="sql_query.php">資料庫操作</a></div>
							 </div>
				 </li>
				 <li>
					 <a title="設備監控" class="btn1" href="index.html">設備監控</a>
							 <div class="headBtn-items">
								 <div><a title= "即時監控" href="update_device.php">即時監控</a></div>
								 <div><a title= "事件監控" href="event.php">事件監控</a></div>
							 </div>
				 </li>
		 </div>
	 </div>
		 <h2>排程備份</h2>
	<hr />

	 備份時間:
	 <select id="time_month" name='time_month'>
	 	<option></option>
	 </select>月
	 <select id="time_day" name='time_day'>
	 	<option></option>
	 </select>日
	 <select id="time_hours" name='time_hours'>
	 	<option></option>
	 </select>時
	 <select id="time_min" name='time_min'>
	 	<option></option>
	 </select>分鐘

    	 <br>
	 備份地點:<select name='device_no' id="dev_no" onchange="showaddr(this.value)">
		  <option value='0'>請選擇</option>
		  <option value='1'>系統本地</option>
		  <option value='2'>遠端主機</option>
	</select >

	<span id="ip_addr"></span>
	<br><br>
	<span id="list"></span>
	<br>

	<hr />
	<div id='dump_file'></div>

</body>


<script >
document.getElementById('time_month').options[0] = new Option('請選擇');
for(var i=1;i<13;i++){
	document.getElementById('time_month').options[i] = new Option(i);
}
document.getElementById('time_month').options[13] = new Option('每月');


document.getElementById('time_day').options[0] = new Option('請選擇');
for(var i=1;i<32;i++){
	document.getElementById('time_day').options[i] = new Option(i);
}
document.getElementById('time_day').options[32] = new Option('每日');

document.getElementById('time_hours').options[0] = new Option('請選擇');
for(var i=1;i<25;i++){
	document.getElementById('time_hours').options[i] = new Option(i-1);
}
document.getElementById('time_hours').options[25] = new Option('每小時');

document.getElementById('time_min').options[0] = new Option('請選擇');
for(var i=1;i<61;i++){
	document.getElementById('time_min').options[i] = new Option(i-1);
}
document.getElementById('time_min').options[61] = new Option('每分鐘');


function showaddr(str){
	if(str === '1'){
	 //document.getElementById("dev_addr").innerHTML = "";
	 //return;
			document.getElementById("ip_addr").innerHTML = "<input type='button' value='備份' onclick='localload()'>";
	 }
	 else if(str === '2'){
				document.getElementById("ip_addr").innerHTML = "<input id='addr' type='text' name='addr'/><input type='button' value='提交' onclick='upload()'>";
			    }
}


function localload(){
	        if (window.XMLHttpRequest) {
			// code for IE7+, Firefox, Chrome, Opera, Safari
	        	xmlhttp = new XMLHttpRequest();
		} else {
		     // code for IE6, IE5
	            xmlhttp = new ActiveXObject("Microsoft.XMLHTTP");
		}
		  xmlhttp.onreadystatechange = function() {
		  if (xmlhttp.readyState == 4 && xmlhttp.status == 200) {
	                document.getElementById("dump_file").innerHTML = xmlhttp.responseText;
		  }
		}
		//xmlhttp.open("GET","crontab_local.php?q="+str,true);
		//xmlhttp.send();
		xmlhttp.open("POST","crontab_local.php",true);
		xmlhttp.setRequestHeader("Content-type","application/x-www-form-urlencoded");
		if(document.getElementById('time_month').value === '每月') M='*';else M=document.getElementById('time_month').value;
		if(document.getElementById('time_day').value === '每日') D='*';else D=document.getElementById('time_day').value;
		if(document.getElementById('time_hours').value === '每小時') H='*';else H=document.getElementById('time_hours').value;
		if(document.getElementById('time_min').value === '每分鐘') m='*';else m=document.getElementById('time_min').value;
		xmlhttp.send("M="+M+"&D="+D+"&H="+H+"&m="+m);
}

function upload(){

	        if (window.XMLHttpRequest) {
			// code for IE7+, Firefox, Chrome, Opera, Safari
	        	xmlhttp = new XMLHttpRequest();
		} else {
		     // code for IE6, IE5
	            xmlhttp = new ActiveXObject("Microsoft.XMLHTTP");
		}
		  xmlhttp.onreadystatechange = function() {
		  if (xmlhttp.readyState == 4 && xmlhttp.status == 200) {
	                //document.getElementById("dev_addr").innerHTML = xmlhttp.responseText;
			 alert("上傳成功");
		  }
		}
		//xmlhttp.open("GET","sql_dump3.php?q="+str,true);
		//xmlhttp.send();
		xmlhttp.open("POST","sql_dump3.php",true);
		xmlhttp.setRequestHeader("Content-type","application/x-www-form-urlencoded");
		xmlhttp.send("addr="+document.getElementById("addr").value);
}


function showlist(){
	if(window.XMLHttpRequest){
			// code for IE7+, Firefox, Chrome, Opera, Safari
	        	xmlhttp2 = new XMLHttpRequest();
		} else {
		     // code for IE6, IE5
	            xmlhttp2 = new ActiveXObject("Microsoft.XMLHTTP");
		}
		  xmlhttp2.onreadystatechange = function() {
		  if (xmlhttp2.readyState == 4 && xmlhttp2.status == 200) {
			document.getElementById("list").innerHTML = xmlhttp2.responseText;
		  }
		}
	//	xmlhttp2.open("GET","monitor_up.php?device_no="+document.getElementById("dev_no").value+"&address="+document.getElementById("dev_addr").value+"&content="+document.getElementById("content").value,true);
	//	xmlhttp2.send();
		xmlhttp2.open("POST","monitor_up.php",true);
		xmlhttp2.setRequestHeader("Content-type","application/x-www-form-urlencoded");
		xmlhttp2.send("device_no="+document.getElementById("dev_no").value+"&address="+document.getElementById("dev_addr").value+"&content="+document.getElementById("content").value);
}

function sleep(milliseconds) {
  var start = new Date().getTime();
    for (var i = 0; i < 1e7; i++) {
        if ((new Date().getTime() - start) > milliseconds){
	      break;
	  }
    }
}
</script>
</html>
